napari/_vispy/vispy_base_layer.py

```diff
@@ -251,43 +220,58 @@ def find_coordinates_of_corners_of_canvas(self):
             tl_raw = [0] * nd
             br_raw = [1] * nd
 
+        # Perform rounding to prevent repeated triggering of requesting
+        # of new pyramid tiles for small camera movements
+        tl_raw = self._rounding * np.floor(np.divide(tl_raw, self._rounding))
+        br_raw = self._rounding * np.ceil(np.divide(br_raw, self._rounding))
```

Comment:
@sofroniewn I don't understand this logic at all... What does the rounding factor represent? How big does a camera movement have to be before we trigger? Possibly we should revert this and leave it for a future PR. It seems to me like there is a real possibility of glitches where we move the camera a little bit and get black canvas around the edges because we haven't reached this threshold.

Reply:
I've made `rounding_factor` an optional kwarg arg to this function and added a very detailed doc string around how it is used, it's a tradeoff - but i think a good one for performance. also the use of `floor` and `ceil` instead of pure rounding means that we always request more than we need and the issued described around the edges cannot happen